open-hackathon-server/src/hackathon/hack/team_manager.py

```diff
@@ -193,17 +193,58 @@ def dismiss_team(self, operator, team_id):
         self.__validate_team_permission(hackathon.id, team, operator)
 
         members = self.db.find_all_objects_by(UserTeamRel, team_id=team.id, status=TeamMemberStatus.Approved)
-        member_ids = [m.user for m in members]
+        member_users = [m.user for m in members]
 
         # delete all team members first
         self.db.delete_all_objects_by(UserTeamRel, team_id=team.id)
         self.db.delete_object(team)
 
-        for u in member_ids:
+        for u in member_users:
             self.create_default_team(hackathon, u)
 
         return ok()
 
+    def quit_team_forcedly(self, team_id, user):
+        """
+        The operator(admin or superadmin) forces a user(team leader or other members) to quit a team.
+        If the user is the only member of the team, the team will be deleted.
+        Else if the user is the leader of a team with several members, the team will be decomposed into several
+        new teams.
+        Else if the user is not the leader of a team with several members, just the user quits the team.
+
+        :rtype: bool
+        :return: if dismiss success, return ok. if not ,return bad request.
+        """
+        team = self.__get_team_by_id(team_id)
```

Comment:
Why not pass `team` object as parameter rather than `team.id`?  Looks like you query the team twice outside and inside method `quit_team_forcedly`